keanu-project/src/main/java/io/improbable/keanu/vertices/bool/nonprobabilistic/BoolProxyVertex.java

```diff
@@ -0,0 +1,53 @@
+package io.improbable.keanu.vertices.bool.nonprobabilistic;
+
+import static io.improbable.keanu.tensor.TensorShapeValidation.checkTensorsMatchNonScalarShapeOrAreScalar;
+
+import com.google.common.collect.Iterables;
+
+import io.improbable.keanu.tensor.Tensor;
+import io.improbable.keanu.tensor.bool.BooleanTensor;
+import io.improbable.keanu.vertices.NonProbabilistic;
+import io.improbable.keanu.vertices.ProxyVertex;
+import io.improbable.keanu.vertices.bool.BoolVertex;
+import io.improbable.keanu.vertices.dbl.KeanuRandom;
+
+public class BoolProxyVertex extends BoolVertex implements ProxyVertex<BoolVertex>, NonProbabilistic<BooleanTensor> {
+
+    /**
+     * This vertex acts as a "Proxy" to allow a BayesNet to be built up before parents are explicitly known (ie for
+     * model in model scenarios) but allows linking at a later point in time.
+     */
+    public BoolProxyVertex() {
```

Comment:
Should we enforce (in the constructor) that a proxy vertex always has a label? Does it make sense to have an unlabelled proxy vertex?

Reply:
Good call - no, Proxy Vertices really do need a label as otherwise they're useless - and making sure people create this as part of the constructor would be the nicest API.